godtossing/Pos/Ssc/GodTossing/Toss/Logic.hs

```diff
@@ -62,11 +61,16 @@ verifyAndApplyGtPayload eoh payload = do
         Right header -> do
             let eos = EpochOrSlot $ Right $ header ^. headerSlotL
             setEpochOrSlot eos
-            let slot = epochOrSlot (const 0) (getSlotIndex . siSlot) eos
-            -- We can freely clear shares after 'slotSecurityParam'
-            -- because it's guaranteed that rollback on more than 'slotSecurityParam'
+            -- We can freely clear shares after 'slotSecurityParam' because
+            -- it's guaranteed that rollback on more than 'slotSecurityParam'
             -- can't happen
-            when (slot >= slotSecurityParam && slot < 2 * slotSecurityParam) resetShares
+            let mkIndex :: SlotAmount -> LocalSlotIndex
+                mkIndex = toEnum . fromIntegral
```

Comment:
I think it would be better to do comparison in `SlotAmount`s, because `toEnum` is unsafe, while conversion from `LocalSlotsIndex` to `SlotAmount` is not.